merge/map.go

```diff
@@ -0,0 +1,18 @@
+package merge
+
+import "maps"
+
+// MapStringString creates a new map and loads it from map args
+// this function takes a least 2 args and later map args take precedence.
```

Comment:
nit: this isn't correct, it can be called with a single argument even if it doesn't make sense to. For example: 

```go
func TestMergeMapStringStringSingle(t *testing.T) {
	m := map[string]string{"a": "a"}
	want := map[string]string{"a": "a"}

	got := MapStringString(m)

	if !reflect.DeepEqual(want, got) {
		t.Errorf("%#v is not equal to %#v", want, got)
	}
}
```
(I guess this is the same thing as the tests with `args` as `nil`? It seems a bit unintuitive to me that nil args would just be ignored, I assume it is because range won't iterate on `nil` maps so `map.Copy` does nothing in this case)

```suggestion
// this function takes a least 1 arg and later map args take precedence.
```

Reply:
The above suggested change may not make sense either; but we should probably have an enforcement mechanism for having two args if we're going to document that it needs it

Reply:
ack: I agree that the description was misleading. I have updated this to be a generic merge function, and also simplified the wording. However, I've left the function signature the same since it should be relatively obvious from it's purpose that supplying 1 map arg (although valid), isn't particularly useful since it would just be a substitute for `maps.Copy`.